express-zod-api/src/documentation-helpers.ts

```diff
@@ -134,59 +98,210 @@ const samples = {
   array: [],
 } satisfies Record<Extract<SchemaObjectType, string>, unknown>;
 
-const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
-const timeRegex = /^\d{2}:\d{2}:\d{2}(\.\d+)?$/;
-
-const getTimestampRegex = (hasOffset?: boolean) =>
-  hasOffset
-    ? /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(([+-]\d{2}:\d{2})|Z)$/
-    : /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z$/;
-
 export const reformatParamsInPath = (path: string) =>
   path.replace(routePathParamsRegex, (param) => `{${param.slice(1)}}`);
 
-export const depictDefault: Depicter = (schema: $ZodDefault, { next }) => ({
-  ...next(schema._zod.def.innerType),
-  default:
-    globalRegistry.get(schema)?.[metaSymbol]?.defaultLabel ||
-    schema._zod.def.defaultValue(),
-});
-
-export const depictAny: Depicter = () => ({ format: "any" });
-
-export const depictUpload: Depicter = ({}: UploadSchema, ctx) => {
-  if (ctx.isResponse)
-    throw new DocumentationError("Please use ez.upload() only for input.", ctx);
-  return { type: "string", format: "binary" };
-};
-
-export const depictFile: Depicter = (schema: FileSchema) => {
-  return {
-    type: "string",
-    format:
-      schema instanceof z.ZodString
-        ? schema._zod.def.checks?.find(
-            (entry) =>
-              isCheck<$ZodStringFormat>(entry, "string_format") &&
-              entry._zod.def.format === "base64",
-          )
-          ? "byte"
-          : "file"
-        : "binary",
-  };
-};
+export const delegate: Depicter = (schema, ctx) => {
+  const result = z.toJSONSchema(schema, {
+    unrepresentable: "any",
+    metadata: globalRegistry,
+    io: ctx.isResponse ? "output" : "input",
+    override: ({ zodSchema, jsonSchema }) => {
```

Comment:
this function is too long and complex.
should split it into multiple ones, similar to previously existed rules